wgsl/index.bs

```diff
@@ -9505,29 +9822,53 @@ value|non-uniform=] during the execution of the function call.
-    * Otherwise, the [=call site tag=] is [=CallSiteNoRestriction=].
-    * For each [=param_i=] in this set, the corresponding [=parameter tag=] is [=ParameterRequiredToBeUniform=].
-    * Remove from the graph all nodes that have been visited.
-* If [=Value_return=] exists, look at which nodes are reachable from it
-    * If this set includes [=MayBeNonUniform=], then the [=function tag=] is [=ReturnValueMayBeNonUniform=].
-    * For each [=param_i=] in this set, the corresponding [=parameter tag=] is [=ParameterRequiredToBeUniformForReturnValue=].
-* For each [=Value_return_i=] node, look at which nodes are reachable from it
-    * If this set includes [=MayBeNonUniform=], the corresponding [=pointer parameter tag=] is [=PointerParameterMayBeNonUniform=].
-    * Otherwise, the corresponding [=pointer parameter tag=] is [=PointerParameterNoRestriction=].
-* If the [=function tag=] has not been assigned, then it is [=NoRestriction=].
-* For each parameter, if it has not been assigned a [=parameter tag=], then it is [=ParameterNoRestriction=].
-
-Note: The entire graph can be destroyed at this point. The tags listed above are all that we need to remember to analyze callers of this function.
+    * The nodes added in this step are called <dfn noexport>interior nodes</dfn>.
+* Initialize as follows:
+    * The [=function tag=] is initialized to [=NoRestriction=].
+    * The [=call site tag=] is initialized to [=CallSiteNoRestriction=].
+    * The [=parameter tag=] for each [=param_i=] is initialized to [=ParameterNoRestriction=].
+    * The [=parameter return tag=] for each [=param_i=] is initialized to [=ParameterReturnNoRestriction=].
+    * The [=pointer parameter tag=] for each [=param_i=], if it exists, is initialized to [=PointerParameterNoRestriction=].
+* For each severity *S* in the order {[=severity/error=], [=severity/warning=], [=severity/info=]}, perform the following:
+    * Let *R.S* be the set of unvisited nodes reachable from [=RequiredToBeUniform.S=].
+    * Mark the [=interior nodes=] in *R.S* as visited.
+    * Let *W* be the [=witness set=] associated with [=RequiredToBeUniform.S=].
+    * If *R.S* includes the node [=MayBeNonUniform=], then trigger a [=uniformity failure=]:
+         * For each [=witness=] *w* in *W*, [=triggered|trigger=] a diagnostic with severity *S*, using the call site and triggering rule from *w*.
+    * Otherwise:
+        * If *R.S* includes [=CF_start=], and the [=call site tag=] has not been updated since initialization, then
+             set the [=call site tag=] to [=CallSiteRequiredToBeUniform.S=], and set its [=witness set=] to *W*.
+        * For each [=param_i=] in *R.S*, if its corresponding [=parameter tag=] has not been updated since initialization, then
+             set that tag to [=ParameterRequiredToBeUniform.S=], and set its [=witness set=] to *W*.
+* Mark all the [=interior nodes=] as unvisited.
```

Comment:
I think it is actually necessary.  The same node might be used to generate a diagnostic due to a call in this function, and also needs to be analyzed for its nonuniform effects on the return value.

Consider this program:
```
@group(0) @binding(0) var t: texture_2d<f32>;
@group(0) @binding(1) var s: sampler;
fn foo(x: vec2<f32>) -> vec2<f32> {
  let c2 = textureSample(t,s,x).xy;
  if (c2.x < 0.0) {
     _ = textureSample(t,s,c2);
  } else {
     return vec2f();
  }
  return c2;
}

@fragment
fn main(@builtin(position) p: vec4<f32>) -> @location(0) vec4<f32> {
  return vec4(foo(p.xy),vec2f());
}
```

The gist of the issue is that the second call to textureSample may be in nonuniform control flow because of the non-uniformity of the c2 value.    
But the non-uniformity of the result also depends on the non-uniformity of the c2 value, because different values are returned from the two different paths out of the "if".

I'm attaching the analysis graph for the foo function.
![tricky](https://user-images.githubusercontent.com/13054574/223232747-85a104a9-f305-4d75-b7a0-a2de82c03f3d.svg)

The second call to textureSample generates an error because there's a path from foo.RequiredToBeUnifom to foo.MayBeNonUniform. Furthermore that path goes through foo.textureSample_return_value, which represents the uniformity summary for the return value of the *first* textureSample call.   In fact that's the only path from RequiredToBeUniform to MayBeNonUniform.  So after generating the diagnostic, the *interior* node foo.textureSample_return_value is marked as unvisited.

Now, observe that *all* paths from `foo.Value_return` to `foo.param_x` actually go through `foo.textureSample_return_value`.  So if we *didn't mark that interior node as unvisited, then we'd miss the fact that the uniformity of the return value depends on the return value of the 'x' parameter.